### ISSUE TITLE
Git, command line, and editor don't know where they are

### ISSUE THREAD

**Issue by rhires** [microsoft/vscode#134036]
Issue Type: <b>Bug</b>

Description first: I was in VSCode, and attempting to `git push`, but could not. I was getting errors about various things, ultimately meaning that I was not pushing from the correct location in my hierarchy. So I went to the command line to attempt to push, and that also failed. I also switched directories using the command line to the correct directory, and still was getting errors about not being able to push. So I  switched my entire window to the correct directory, then I could do the push from the CL with no problems. 

I have a Visual Studio solution in VSCode, which VSCode does not treat the same way as Visual Studio does, which leads to confusion about which part of the directory structure actually will respond correctly to git commands via the command line. 

So in my case, from the editor window I'm in a directory above the .sln file, and two directories above the project directory. From this location, I can't do a git push (via the UI) because the project directory is what needs to be pushed. So I switched to the terminal, and cd solution.slnDirectory/ProjectDirectory and then run git, and it still fails. 

Then I change the editor so that the top level directory is the ProjectDirectory, and then I was able to push via the command line. 

So basically, when I run git from the command line, git thinks I'm at the top level window in the UI, even though I'm not at the level in the command line. 

VS Code version: Code 1.60.2 (7f6ab5485bbc008386c4386d08766667e155244e, 2021-09-22T11:59:27.195Z)
OS version: Darwin x64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.04GB free)|
|Process Argv|--crash-reporter-id c334bbd5-69b6-4ee0-9bdf-260924daed9c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-icalendar|af4|1.0.1
ng-template|Ang|12.2.0
swagger-viewer|Arj|3.1.2
npm-intellisense|chr|1.4.0
jshint|dba|0.11.0
vscode-eslint|dba|2.1.25
githistory|don|0.6.18
gitlens|eam|11.6.0
EditorConfig|Edi|0.16.4
tslint|eg2|1.0.47
vscode-npm-script|eg2|0.3.22
prettier-vscode|esb|9.0.0
nasc-touchbar|fel|1.12.0
git-project-manager|fel|1.7.1
cshtml-tm|fir|1.0.0
remotehub|Git|0.14.0
gc-excelviewer|Gra|3.0.44
beautify|Hoo|1.5.0
rest-client|hum|0.24.5
csharpextensions|jch|1.3.1
angular-essentials|joh|12.0.0
Angular2|joh|12.0.0
vscode-peacock|joh|3.9.1
Angular-BeastCode|Mik|10.0.3
prettify-json|moh|0.0.3
csharp|ms-|1.23.15
mssql|ms-|1.10.1
remote-ssh|ms-|0.65.8
remote-ssh-edit|ms-|0.65.8
vscode-typescript-tslint-plugin|ms-|1.3.3
debugger-for-chrome|msj|4.13.0
debugger-for-edge|msj|1.0.15
angular2-inline|nat|0.0.17
angular-console|nrw|17.11.2
material-icon-theme|PKi|4.10.0
vscode-commons|red|0.0.6
vscode-yaml|red|0.22.0
yo|sam|0.9.3
aspnet-helper|sch|0.6.4
autoimport|ste|1.5.4

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551:30345470
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30358480
pythondataviewer:30285071
pythonvsuse255:30340121
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
vsccppwt:30364497
pythonvssor306:30344512
bridge0708:30335490
pygetstartedt2:30371810
dockerwalkthrucf:30370837
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedc:30364665
pythonvspyt187:30373474
pydsgsc2:30361791

```

</details>

**Comment by rhires** [comment]
Hey @lszomoru - thanks for having a look at this. It looks like I started with the built in git commands in VS Code, and then switched to the integrated terminal. I guess my expectation is that VS Code will not make git available in the UI if you're in the "wrong" folder - one that does not have git enabled. But the VSC git in the UI didn't "know" that the folder it was in didn't have git enabled at that point in the hierarchy, so it just failed. 

Example: 
Editor window - shows MySource as the top level directory. Git commands are available in the UI.
Below MySource is another folder - MySolution - which holds MySolution.sln
Below MySolution is the MyProject Folder which holds the .csproj file. This is where git is enabled. This is the directory I had to use the CLI to cd to for git to work, from the CLI, as it didn't work from the UI built-in commands. 

I hope that isn't too confusing.
